### PR TITLE
[INLONG-2384][SDK] Sort-sdk support Tube consumer of PB compression cache message protocol

### LINKED ISSUE #2384: [Feature] Sort-sdk support Tube consumer of PB compression cache message protocol.
### Description

 Sort-sdk support Tube consumer of PB compression cache message protocol.
Parent issue: #2375 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wardlican**
…message protocol

[INLONG-2384] Sort-sdk support Tube consumer of PB compression cache message protocol

### Title Name: [INLONG-2384][SDK] Sort-sdk support Tube consumer of PB compression cache message protocol

where *XYZ* should be replaced by the actual issue number.

Fixes #2384

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java [hunk 1]
```diff
@@ -90,15 +89,15 @@ public boolean init(Object object) {
...
-            try {
-                if (messageConsumer == null) {
-                    context.getStatManager().getStatistics(context.getConfig().getSortTaskId(),
-                            inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
-                            .addAckFailTimes(1L);
-                    LOG.error("consumer == null");
-                    return;
-                }
+            if (messageConsumer == null) {
+                context.getStatManager().getStatistics(context.getConfig().getSortTaskId(),
```
Comment (imvan) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java:93]: need new line when call getStatistics()

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java [hunk 2]
```diff
@@ -174,6 +174,15 @@ public long getAckedOffset() {
         return 0L;
     }
 
+    /**
+     * isValidState
+     */
+    public void isValidState() {
```
Comment (imvan) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java:180]: why not return false, but throw exception in the method called isxxx when the state is not valid.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java (hunk 1 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java:93]: ok

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java (hunk 2 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java:180]: This method has not been completed yet

**Review by imvan — APPROVED** [review]

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java [hunk 3]
```diff
@@ -261,11 +275,14 @@ public void run() {
...
-                            msgs.add(new InLongMessage(msg.getData(), getAttributeMap(msg.getAttribute())));
+                            List<InLongMessage> deserialize = deserializer
+                                    .deserialize(context, inLongTopic, getAttributeMap(msg.getAttribute()),
+                                            msg.getData());
+                            msgs.addAll(deserialize);
                             context.getStatManager()
                                     .getStatistics(context.getConfig().getSortTaskId(),
                                             inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
-                                    .addMsgCount(1L).addConsumeSize(msg.getData().length);
+                                    .addMsgCount(deserialize.size()).addConsumeSize(msg.getData().length);
```
Comment (luchunliang) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java:285]: Why do the code put deserialize.size() to MsgCount? "deserialize.size()" is the message count or the uncompressed message size.

**Review by luchunliang — APPROVED** [review]

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java (hunk 3 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java:285]: "Size" represents the size of the list, that is, how many messages there are

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**